totalRP3/Core/Color.lua

```diff
@@ -604,7 +606,7 @@ function TRP3_API.GenerateReadableColor(foregroundColor, backgroundColor, target
 	local FgYs = CalculateLuminance(FgR, FgG, FgB);
 	local FgLc = CalculateLightnessContrast(FgYs, BgYs);
 
-	if math.abs(FgLc) < targetContrast then
```

Comment:
Note; only reason for this change is to just skip the contained loop which attempts to brighten the color until it meets the target contrast.

As the contrast for monochromatic mode is infinity, the loop is pointless and we can bypass it and just go straight to the snapping logic right afterwards.